File: func17.js
// func17

const add = x => (y => x + y)

const inc = add(1)

// const result = inc(2)

const modulo = dvr => dvd => dvd % dvr

const isOdd = modulo(2)

const filter = pred => xs => xs.filter(pred)

const allOdds = filter(isOdd)

// const result = allOdds([1, 2, 3, 4])

const replace = regex => repl => str =>
    str.replace(regex, repl)

const censor = replace(/[aeiou]/ig)('*')

// const result = censor('hello world')

const map = f => xs => xs.map(f)

const censorAll = map(censor)

const result = censorAll(['Hello', 'World'])

console.log(result)

File: func17.py
# func17
import re

modulo = lambda x: lambda y: y % x

is_odd = modulo(2)

filt = lambda pred: lambda xs: filter(pred, xs)

all_odds = filt(is_odd)

# result = all_odds([1, 2, 3, 4])

replace = lambda regex: lambda repl: lambda st: re.sub(regex, repl, st)

censor = replace('[aeiou]')('*')

# result = censor('hello world')

custom_map = lambda f: lambda xs: map(f, xs)

censor_all = custom_map(censor)

result = list(censor_all(['hello', 'world']))

# print(list(result))
print(result)